pipelines/resource_pipeline.go

```diff
@@ -7,319 +7,188 @@ import (
 
-	var resp createPipelineResponse
-	err := a.client.Post(a.ctx, "/pipelines", s, &resp)
+	createdPipeline, err := w.Pipelines.Create(ctx, createPipelineRequest.CreatePipeline)
 	if err != nil {
-		return "", err
+		return err
 	}
-	id := resp.PipelineID
-	err = a.waitForState(id, timeout, StateRunning)
+	var id string
+	if d.Get("dry_run").(bool) {
+		id = createdPipeline.EffectiveSettings.Id
+		d.SetId(id)
+		return nil
+	} else {
+		id = createdPipeline.PipelineId
+	}
+	err = waitForState(w, ctx, id, timeout, pipelines.PipelineStateRunning)
 	if err != nil {
 		log.Printf("[INFO] Pipeline creation failed, attempting to clean up pipeline %s", id)
-		err2 := a.Delete(id, timeout)
+		err2 := Delete(w, ctx, id, timeout)
 		if err2 != nil {
 			log.Printf("[WARN] Unable to delete pipeline %s; this resource needs to be manually cleaned up", id)
-			return "", fmt.Errorf("multiple errors occurred when creating pipeline. Error while waiting for creation: \"%v\"; error while attempting to clean up failed pipeline: \"%v\"", err, err2)
+			return fmt.Errorf("multiple errors occurred when creating pipeline. Error while waiting for creation: \"%v\"; error while attempting to clean up failed pipeline: \"%v\"", err, err2)
 		}
 		log.Printf("[INFO] Successfully cleaned up pipeline %s", id)
-		return "", err
+		return err
 	}
-	return id, nil
+	d.SetId(id)
+	return nil
 }
 
-func adjustForceSendFields(s *PipelineSpec) {
-	for i := range s.Clusters {
-		cluster := &s.Clusters[i]
-		// TF Go SDK doesn't differentiate between the default and not set values.
-		// If nothing is specified, DLT creates a cluster with enhanced autoscaling
-		// from 1 to 5 nodes, which is different than sending a request for zero workers.
-		// The solution here is to look for the Spark configuration to determine
-		// if the user only wants a single node cluster (only master, no workers).
-		if cluster.SparkConf["spark.databricks.cluster.profile"] == "singleNode" {
-			cluster.ForceSendFields = append(cluster.ForceSendFields, "NumWorkers")
-		}
-	}
+func Read(w *databricks.WorkspaceClient, ctx context.Context, id string) (*pipelines.GetPipelineResponse, error) {
+	return w.Pipelines.Get(ctx, pipelines.GetPipelineRequest{
+		PipelineId: id,
+	})
 }
 
-func (a PipelinesAPI) Read(id string) (p PipelineInfo, err error) {
-	err = a.client.Get(a.ctx, "/pipelines/"+id, nil, &p)
-	return
-}
+func Update(w *databricks.WorkspaceClient, ctx context.Context, d *schema.ResourceData, timeout time.Duration) error {
+	var updatePipelineRequest updatePipelineRequestStruct
+	common.DataToStructPointer(d, pipelineSchema, &updatePipelineRequest)
+	updatePipelineRequest.EditPipeline.PipelineId = d.Id()
+	adjustForceSendFields(&updatePipelineRequest.Clusters)
 
-func (a PipelinesAPI) Update(id string, s PipelineSpec, timeout time.Duration) error {
-	adjustForceSendFields(&s)
-	err := a.client.Put(a.ctx, "/pipelines/"+id, s)
+	err := w.Pipelines.Update(ctx, updatePipelineRequest.EditPipeline)
```

Comment:
Does `dry_run` logic applies here? What if the user changes the `dry_run` option?

Reply:
The `dry_run` doesn't affect the update call

Reply:
but really - do we need to support `dry_run` at all?

Reply:
I just learned what it is (I was asking above). And I don't think we need to support it.